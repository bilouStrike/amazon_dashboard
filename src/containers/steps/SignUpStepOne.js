import React from "react";
import { Button, Input, Form } from "antd";
import IntlMessages from "util/IntlMessages";

const SignUpStepOne = ({ setStep, setStepOneValues, stepOneValues }) => {

    const nextStep = async (values) => {
        await setStepOneValues(values);
        setStep(2);
    }

    return (
            <Form
                onFinish={nextStep}
                initialValues={{ 
                    FirstName: stepOneValues.FirstName,
                    LastName: stepOneValues.LastName,
                    email: stepOneValues.email,
                    username: stepOneValues.username,
                    password: stepOneValues.password,
                    confirm: stepOneValues.confirm,
                }}
                name="basic"
                className="gx-signin-form gx-form-row0">
                <Form.Item rules={[
                    { required: true, message: 'Please input your First Name!'},
                    { pattern: new RegExp(/^[A-Za-z ]+$/), message: 'First Name should be alphabetic'}]
                  } 
                    name="FirstName"
                >
                  <Input placeholder="First Name" className="amazon-tool-home-input"/>
                </Form.Item>

                <Form.Item rules={[
                    { required: true, message: 'Please input your Last Name!'},
                    { pattern: new RegExp(/^[A-Za-z ]+$/), message: 'Last Name should be alphabetic'}]
                  } 
                    name="LastName"
                >
                  <Input placeholder="Last Name" className="amazon-tool-home-input"/>
                </Form.Item>

                <Form.Item
                    rules={[{ required: true, message: 'Please input your email!' },
                    { type: 'email', message: 'This is not valid email!' }]}
                    name="email"
                >
                    <Input placeholder="Email" className="amazon-tool-home-input"/>
                </Form.Item>

                <Form.Item rules={[
                    { required: true, message: 'Please chose your username!'},
                    { min: 3, message: 'Username should be at least 4 characters!'},
                    { max: 15, message: 'Username should be maximun of 15 characters!'},
                    { pattern: new RegExp(/^[a-zA-Z0-9]+$/), message: 'Characters allowed a-z and 0-9.'}]}
                    name="username"
                >
                  <Input placeholder="Username" className="amazon-tool-home-input"/>
                </Form.Item>
                
                <Form.Item
                  name="password"
                  rules= {[
                    { required: true, message: 'Please input your Password!'},
                    { min: 8 , message: 'The string must contain at least 8 numeric character'}]}  
                >
                    <Input type="password" placeholder="Password" className="amazon-tool-home-input"/>
                </Form.Item>

                <Form.Item
                  name="confirm"
                  dependencies={['password']}
                  rules={[
                    {
                      required: true,
                      message: 'Please confirm your password!',
                    },
                    ({ getFieldValue }) => ({
                      validator(rule, value) {
                        if (!value || getFieldValue('password') === value) {
                          return Promise.resolve();
                        }
                        return Promise.reject('The two passwords that you entered do not match!');
                      },
                    }),
                  ]}
                >
                  <Input type="password" placeholder="Confirm password" className="amazon-tool-home-input"/>
                </Form.Item>

                <Form.Item>
                  <Button type="primary" className="gx-mb-0" style={{height:'48px'}} size="large" block htmlType="submit">
                    <IntlMessages id="app.userAuth.NextStep"/>
                  </Button>
                </Form.Item>
            </Form>
    )
}
export default SignUpStepOne;