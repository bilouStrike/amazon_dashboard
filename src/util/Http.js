import axios from 'axios';
import { BASE_API_URL } from 'constants/Common';

export default axios.create({
  baseURL: BASE_API_URL,
  headers: {
    'Content-Type': 'application/json',
  }
});
