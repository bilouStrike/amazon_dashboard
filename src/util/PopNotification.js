import { message } from "antd";

const showMessahe = (status, response) => {
    
    message.destroy();
    let config = {
        content:response,
        duration: 5,
        style: { padding:'120px', marginTop:'5%' }
    }

    if( status === 'success' ) {
        message.success(config);
        return;
    }
    message.error(config);
}
export default showMessahe;