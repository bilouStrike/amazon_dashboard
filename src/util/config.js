module.exports = {
  footerText: 'Copyright Amazon Tool Â© 2020',
}
