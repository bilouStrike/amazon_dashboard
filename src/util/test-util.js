// test-utils.js
import React from 'react'
import { render } from '@testing-library/react'
/*import { ThemeProvider } from 'my-ui-lib'
import { TranslationProvider } from 'my-i18n-lib'
import defaultStrings from 'i18n/en-x-default'*/
import { IntlProvider } from "react-intl";

const AllTheProviders = ({ children }) => {
  return (
    <IntlProvider locale="en">
        {children}
    </IntlProvider>
  )
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }