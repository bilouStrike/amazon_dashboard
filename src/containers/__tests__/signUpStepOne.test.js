import React from "react";
import "@testing-library/jest-dom/extend-expect";
import '../../util/testConfig';
import { render, fireEvent, cleanup } from '../../util/test-util';

import SignUpStepOne from '../steps/SignUpStepOne';

afterEach(cleanup);

it("should render signinstepOne component", () => {
    
  const mockStepOneValues = {};
  const mockSetStepOneValues = jest.fn();
  const mockSetStep = jest.fn();

  const { asFragment } = render(
    <SignUpStepOne 
        stepOneValues={mockStepOneValues}
        setStepOneValues={mockSetStepOneValues}
        setStep={mockSetStep}/>
    );
  expect(asFragment()).toMatchSnapshot();
});
