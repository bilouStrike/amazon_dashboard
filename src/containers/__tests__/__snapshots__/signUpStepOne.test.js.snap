// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render signinstepOne component 1`] = `
<DocumentFragment>
  <form
    class="ant-form ant-form-horizontal gx-signin-form gx-form-row0"
    id="basic"
  >
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input amazon-tool-home-input"
              id="basic_FirstName"
              placeholder="First Name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input amazon-tool-home-input"
              id="basic_LastName"
              placeholder="Last Name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input amazon-tool-home-input"
              id="basic_email"
              placeholder="Email"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input amazon-tool-home-input"
              id="basic_username"
              placeholder="Username"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input amazon-tool-home-input"
              id="basic_password"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input amazon-tool-home-input"
              id="basic_confirm"
              placeholder="Confirm password"
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <button
              class="ant-btn gx-mb-0 ant-btn-primary ant-btn-lg ant-btn-block"
              style="height: 48px;"
              type="submit"
            >
              app.userAuth.NextStep
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
